presto-orc/src/main/java/com/facebook/presto/orc/writer/FloatColumnWriter.java

```diff
@@ -107,14 +107,17 @@ public void writeBlock(Block block)
         }
 
         // record values
+        int nonNullValueCount = 0;
         for (int position = 0; position < block.getPositionCount(); position++) {
             if (!block.isNull(position)) {
                 int intBits = (int) type.getLong(block, position);
                 float value = intBitsToFloat(intBits);
                 dataStream.writeFloat(value);
                 statisticsBuilder.addValue(value);
+                nonNullValueCount++;
             }
         }
+        return nonNullValueCount * Float.BYTES + (block.getPositionCount() - nonNullValueCount) * NULL_SIZE;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).